### ISSUE TITLE
[wg/css] CSS Working Group Charter

### ISSUE THREAD

**Issue by svgeesus** [w3c/strategy#477]
New charter proposal, reviewers please take note.

# Charter Review

[Charter:](https://w3c.github.io/charter-drafts/2025/css-wg.html)

If applicable:
[diff from previous charter](https://services.w3.org/htmldiff?doc1=https%3A%2F%2Fwww.w3.org%2F2023%2F01%2Fcss-2023.html&doc2=https%3A%2F%2Fw3c.github.io%2Fcharter-drafts%2F2025%2Fcss-wg.html) and [diff from template](https://services.w3.org/htmldiff?doc1=https%3A%2F%2Fw3c.github.io%2Fcharter-drafts%2Fcharter-template.html&doc2=https%3A%2F%2Fw3c.github.io%2Fcharter-drafts%2F2025%2Fcss-wg.html)

[chair dashboard](https://www.w3.org/PM/Groups/chairboard.html?gid=wg/css)

What kind of charter is this? Check the relevant box / remove irrelevant branches. 

* Existing 
 - [x] Existing WG recharter 

Communities suggested for outreach: 

Known or potential areas of concern: 

Where would charter proponents like to see issues raised? [CSSWG issue](https://github.com/w3c/csswg-drafts/issues/10671)

Anything else we should think about as we review? 

Note: proposed chairs should be copied @... on this issue.

@astearns @atanassov

**Comment by ruoxiran** [comment]
There is an extra < in section 1. There are no other comments from APA.

**Comment by svgeesus** [comment]
Thanks, typo fixed

**Comment by svgeesus** [comment]
@simoneonofri any security comments? @plehegar any comments from PING?

**Comment by svgeesus** [comment]
@simoneonofri thank you. As these are comments on individual deliverables it would be helpful to raise them on csswg repo against the relevant spec, so that they don't get lost.

On your more general points:

> “introduces no new privacy leaks, or security considerations,”

I _hate_ when people write that, such a bold and unsubstantiated claim. Instead I encourage 

> No privacy concerns have been raised against this specification
> No security concerns have been raised against this specification

In other words statements which are provably true or false.

Some old/unmaintained specs still have one section for both Privacy and Security, or indeed no section at all; I fix these as they get published to /TR.

**Comment by simoneonofri** [comment]
@svgeesus thank you for your answer.

In general, I think that raising concerns, it seems more an "external" think, even if this should be a joint work between the specs developer and security experts.

And it can make sense that there are no _new_ considerations, but it can be good a reference to the old one.

**Comment by svgeesus** [comment]
@tjwhalen any comments on this draft charter from a privacy perspective?

**Comment by plehegar** [comment]
PING is fine.

**Comment by plehegar** [comment]
(from PING) There has a batch of [issues related to privacy](https://github.com/w3cping/tracking-issues/labels/wg%3Acss).  Was there progress by the CSS WG? In particular on the font fingerprinting front.
Also, media queries did not have a privacy review since 2020. What's the plan to move media queries forward?

**Comment by tjwhalen** [comment]
> @tjwhalen any comments on this draft charter from a privacy perspective?
To build on @plehegar's comments, based on my quick read and helpful discussion from #privacy-reviews (thx Jeffrey Yasskin):
- given the concerns around outstanding privacy issues, the charter should be explicit about planned work: which modules do you expect to work on, create new levels of, and/or advance to CR?
- the charter also should establish a link between the "Classification" in css-2024 through css-2026 and the ED/WC/CR state of the associated specification. In regard to media queries (mentioned above): https://drafts.csswg.org/css-2024/#experimental indicates that most non-CR features aren't intended to be released widely, but it seems to be out of date for media queries: https://caniuse.com/?search=prefers

Thanks for taking time to solicit my comments, particularly as they're at a late stage (and I may be lacking some context, having just shown up!).

**Comment by simoneonofri** [comment]
> @simoneonofri thank you. As these are comments on individual deliverables it would be helpful to raise them on csswg repo against the relevant spec, so that they don't get lost.

we can catch during the various transition requests
 
> On your more general points:
> 
> > “introduces no new privacy leaks, or security considerations,”
> 
> I _hate_ when people write that, such a bold and unsubstantiated claim. Instead I encourage
> 
> > No privacy concerns have been raised against this specification
> > No security concerns have been raised against this specification
> 
> In other words statements which are provably true or false.

In general, the issue is, has anyone asked within the WG what are the Privacy and Security Considerations? 
It's not easy when you're in the “I'm building something” mode to figure out “what could go wrong,” but in general on the CSS side there's a lot of fingerprinting, as they also pointed out from PING and that also has an impact often at the Security level, particularly when we go to work on the GPU.
Probably a correct statement could be “According to the Threat Model (link to threat model), these are the residual threats we have, or I don't sound new threats compared to the general model.”
Step by step we will get there :)
 
> Some old/unmaintained specs still have one section for both Privacy and Security, or indeed no section at all; I fix these as they get published to /TR.

Thank you!

**Comment by svgeesus** [comment]
> Probably a correct statement could be “According to the Threat Model (link to threat model), these are the residual threats we have, or I don't sound new threats compared to the general model.”

I didn't follow the "I don't sound" part. Autocorrect?

**Comment by simoneonofri** [comment]
s/sound/found/ :)

**Comment by astearns** [comment]
@svgeesus we resolved yesterday to adopt intersection observer

https://log.csswg.org/irc.w3.org/css/2024-09-26/#e1651268

Could you add this to the upcoming charter?

**Comment by astearns** [comment]
(support for this move: https://github.com/w3c/strategy/issues/457#issuecomment-2369521052 )

**Comment by svgeesus** [comment]
> Could you add this to the upcoming charter?

Will do!

**Comment by plehegar** [comment]
Several of us sat down last week to figure out how to move forward on [fonts fingerprinting issue](https://github.com/w3c/csswg-drafts/issues/5421) (see also [Fonts, Privacy, and Not Breaking the Web](https://www.w3.org/2024/09/font-i18n-privacy.html)). @svgeesus agreed to make a new pull request to attempt to resolve the matter. However, I don't believe the charter has to block on resolving this issue.

**Comment by plehegar** [comment]
> > However, I don't believe the charter has to block on resolving this issue.
> 
> In that case, @plehegar for this charter review, can we mark the Privacy review as complete?

I don't expect anything else to come from PING at this point. My suggestion would be that we mention this ongoing work when starting the AC review.

**Comment by svgeesus** [comment]
Charter now has [current deliverables list](https://w3c.github.io/charter-drafts/2025/css-wg.html#deliverables) including [Intersection Observer](https://w3c.github.io/charter-drafts/2025/css-wg.html#IntersectionObserver)

**Comment by plehegar** [comment]
from a Privacy perspective, there is still a concern on the lack of progress on the font fingerprinting issue.  looking forward for progress following conversations at TPAC.

**Comment by plehegar** [comment]
one minor bit: the text "In order to advance to [Proposed Recommendation ](https://www.w3.org/policies/process/#RecsPR)," should be updated with the latest charter template. (there might be other things as well)

**Comment by svgeesus** [comment]
> one minor bit: the text "In order to advance to [Proposed Recommendation ](https://www.w3.org/policies/process/#RecsPR)," should be updated with the latest charter template.

Fixed in https://github.com/w3c/charter-drafts/commit/4db4b61848ee7137d08e3ab0955104ed24c39c9a

**Comment by siusin** [comment]
It'd be nice to record changes in this update in the [charter history](https://w3c.github.io/charter-drafts/2025/css-wg.html#about) table.

**Comment by plehegar** [comment]
[VOTE by 2025-01-10/11: Proposed Charter for the CSS Working Group; Current Charter Extended](https://lists.w3.org/Archives/Member/w3c-ac-members/2024OctDec/0054.html)

**Comment by plehegar** [comment]
We received a formal objection but it will likely get withdrawn.

**Comment by svgeesus** [comment]
[FO withdrawn today](https://lists.w3.org/Archives/Public/public-review-comments/2025Mar/0000.html)

**Comment by svgeesus** [comment]
New charter announced today
